crates/bevy_picking/src/input.rs

```diff
@@ -0,0 +1,264 @@
+//! This module provides unsurprising default inputs to `bevy_picking` through [`PointerInput`].
+//! The included systems are responsible for sending  mouse and touch inputs to their
+//! respective `Pointer`s.
+//!
+//! Because this has it's own plugin, it's easy to omit it, and provide your own inputs as
+//! needed. Because `Pointer`s aren't coupled to the underlying input hardware, you can easily mock
+//! inputs, and allow users full accessibility to map whatever inputs they need to pointer input.
+//!
+//! If, for example, you wanted to add support for VR input, all you need to do is spawn a pointer
+//! entity with a custom [`PointerId`], and write a system
+//! that updates its position. If you want this to work properly with the existing interaction events,
+//! you need to be sure that you also write a [`PointerInput`] event stream.
+
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_input::touch::{TouchInput, TouchPhase};
+use bevy_input::{prelude::*, ButtonState};
+use bevy_math::Vec2;
+use bevy_reflect::prelude::*;
+use bevy_render::camera::RenderTarget;
+use bevy_utils::{tracing::debug, HashMap, HashSet};
+use bevy_window::{PrimaryWindow, WindowEvent, WindowRef};
+
+use crate::{
+    pointer::{Location, PointerAction, PointerButton, PointerId, PointerInput, PressDirection},
+    PointerBundle,
+};
+
+use crate::PickSet;
+
+/// Common imports for `bevy_picking`.
+pub mod prelude {
+    pub use crate::input::InputPlugin;
+}
+
+/// Adds mouse and touch inputs for picking pointers to your app. This is a default input plugin,
+/// that you can replace with your own plugin as needed.
+///
+/// [`crate::PickingPluginsSettings::is_input_enabled`] can be used to toggle whether
+/// the core picking plugin processes the inputs sent by this, or other input plugins, in one place.
+#[derive(Copy, Clone, Resource, Debug, Reflect)]
+#[reflect(Resource, Default)]
+pub struct InputPlugin {
```

Comment:
We should rename this: it conflicts with `bevy_input::InputPlugin` in a confusing way.

Reply:
`PickingInputPlugin` seems fine.

Reply:
Going with `PointerInputPlugin` to match the `PointerInput` event, which it provides.